src/battle_tower.c

```diff
@@ -147,7 +147,7 @@ const u16 gBattleFrontierHeldItems[] =
 #include "data/battle_frontier/battle_frontier_trainers.h"
 #include "data/battle_frontier/battle_frontier_mons.h"
 
-const u8 gTowerMaleFacilityClasses[] = 
+const u8 gTowerMaleFacilityClasses[30] =
```

Comment:
Is there any point to specifying the length of the arrays `gTowerMaleFacilityClasses`, `gTowerFemaleFacilityClasses`, `gTowerMaleTrainerGfxIds`, and `gTowerFemaleTrainerGfxIds`? The compiler should be able to figure that out automatically.

Reply:
yeah, they should be removed.  They are only useful when declaring them as extern.

Reply:
They have to be there, otherwise you won't be able to extern it while specyfing the array length.

Reply:
What do you mean? If you need to extern it from another file then you can do so without specifying the length.

Reply:
Apparently removing these causes Travis to have an internal compiler error?

Reply:
I guess it has something to do with apprentice.c using ARRAY_COUNT on this array but it not being defined in that file so the length needs to be explicit.